src/fluree/db/ledger/consensus/update_state.clj

```diff
@@ -72,31 +72,43 @@
                              current-val)))]
     (= proposed-v (get-in new-state ks))))
 
+(defn- extract-flake-object
+  "Returns flake object (.-o flake) from block-map of the given type whose
+  subject matches (:t tx-data). Returns nil if none is found."
+  [{:keys [flakes] :as _block-map} {:keys [t] :as _tx-data} type]
+  (some #(let [^Flake flake %]
```

Comment:
you can use `flake/s`, `flake/p`, and `flake/o` below to avoid needing to carry around these type hints.